src/QueryFilterHelpers.ts

```diff
@@ -0,0 +1,604 @@
+  for (let i = 0; i < library.library.statements.def.length; i++) {
+    const statement = library.library.statements.def[i];
+    const expression = findClauseInExpression(statement.expression, localId);
+    if (expression) {
+      return expression;
+    }
+  }
+  return null;
+}
+
+/**
+ * Recursively search an ELM tree for an expression (clause) with a given localId.
+ *
+ * @param expression The expression tree to search for the clause in.
+ * @param localId The localId to look for.
+ * @returns The expression if found or null.
+ */
+function findClauseInExpression(expression: any, localId: string): ELMExpression | null {
+  if (typeof expression === 'string' || typeof expression === 'number' || typeof expression === 'boolean') {
+    return null;
+  } else if (Array.isArray(expression)) {
+    for (let i = 0; i < expression.length; i++) {
+      const memberExpression = findClauseInExpression(expression[i], localId);
+      if (memberExpression) {
+        return memberExpression as ELMExpression;
+      }
+    }
+    return null;
+  } else if (expression.localId == localId) {
+    return expression as ELMExpression;
+  } else {
+    return findClauseInExpression(Object.values(expression), localId);
+  }
+}
+
+/**
+ * Parse information about the sources in a given query.
+ *
+ * @param query The Query expression to parse.
+ * @returns Information about each source. This is usually an array of one.
+ */
+function parseSources(query: ELMQuery): SourceInfo[] {
+  const sources: SourceInfo[] = [];
+  query.source.forEach(source => {
+    if (source.expression.type == 'Retrieve') {
+      const sourceInfo: SourceInfo = {
+        sourceLocalId: source.localId,
+        retrieveLocalId: source.expression.localId,
+        alias: source.alias,
+        resourceType: parseDataType(source.expression as ELMRetrieve)
+      };
+      sources.push(sourceInfo);
+    }
+  });
+  return sources;
+}
+
+/**
+ * Pulls out the resource type of the retrieve.
+ *
+ * @param retrieve The retrieve expression to pull out resource type from.
+ * @returns FHIR ResourceType name.
+ */
+function parseDataType(retrieve: ELMRetrieve): string {
+  return retrieve.dataType.replace(/^(\{http:\/\/hl7.org\/fhir\})?/, '');
+}
+
+/**
+ * Interprets an expression into a filter tree. This is the central point where the interpreting occurs. This function
+ * determines the expression type and sends it to the correct place to be parsed.
+ *
+ * @param expression The ELM expression/clause to attempt to interpret into a filter.
+ * @param library The ELM library, in case it is needed for calculating intervals.
+ * @param parameters The parameters used for calculation.
+ * @returns The simpler Filter representation of this clause.
+ */
+function interpretExpression(expression: ELMExpression, library: ELM, parameters: any): AnyFilter {
+  switch (expression.type) {
+    case 'FunctionRef':
+      //interpretFunctionRef(expression as ELMFunctionRef, library);
```

Comment:
Update: noticed that the function is indeed called by other interpreters, but the question still stands for why this is commented

Reply:
So functions could have a few different actions:
- Ensuring a specific data type of an attribute, transforming if necessary. ex. Global."Normalize Interval"
- Transforming the type from a FHIR type to a CQL system type. ex. FHIRHelpers."ToString". 
- Actually doing some filtering on something.

The former two situations are called in the other interpret functions to see if know this function can be treated as a passthrough to accessing an attribute on a source resource.

This commented out code is for the later situation, function doing some filtering, which I haven't done any work to support yet. I think this should actually a different function that does this work.

I'll remove this case statement because it's just not supported right now.

Reply:
Removed this case in 6ce9a6a. Will update the `interpretFunctionRef` function name/comment to better explain its purpose to see if we can just consider it a passthrough to the operand.